src/alpaka/Makefile

```diff
@@ -4,6 +4,10 @@ TARGET := $(BASE_DIR)/$(TARGET_NAME)
 include Makefile.deps
 EXTERNAL_DEPENDS := $(alpaka_EXTERNAL_DEPENDS)
 
+ifneq ($(wildcard $(CUDA_BASE)),)
+EXTERNAL_DEPENDS += CUDA
+endif
```

Comment:
Please move this to `Makefile.deps`, otherwise the main `Makefile` won't be aware of it (although with system dependencies it makes little difference in practice).

You could also use `ifdef CUDA_BASE` for the check (that is used already elsewhere). The main `Makefile` leaves `CUDA_BASE` undefined if it detects CUDA doesn't exist.

(I'll look the rest later today, they need a bit more thought)